packages/react-components/react-virtualizer/src/components/Virtualizer/Virtualizer.types.ts

```diff
@@ -139,6 +145,28 @@ export type VirtualizerConfigProps = {
    * Virtualizer context can be passed as a prop for extended class use
    */
   virtualizerContext?: VirtualizerContextProps;
+
+  /*
+   * Callback object to notify when a flagged index has been rendered
+   */
+  flagIndex?: FlaggedIndexCallback;
+
+  /**
+   * Imperative ref contains our scrollTo index functionality for user control.
+   */
+  imperativeSizeCallbacks?: RefObject<ImperativeSizeCallbacks>;
+};
+
+export type FlaggedIndexCallback = {
+  /**
+   * Callback for notifying when a flagged index has been rendered
+   */
+  onRenderedFlaggedIndex: (index: number) => number;
+
+  /*
+   * Callback for notifying when a flagged index has been rendered
+   */
+  flaggedIndex: React.MutableRefObject<number | null>;
```

Comment:
I see that internally you overwrite the value of this ref, this could be confusing for users, since it's not like a DOM ref that needs to be set by the component. In this case the user sets an explicit value which gets wiped by the component. I don't think it's necessarily that bad, I just have never seen any other ref usage do this.

Would it be possible to also make this set imperatively? like `virtualizerRef.setFlaggedIndex()` ?